client_test.go

```diff
@@ -525,6 +566,10 @@ func buildClient(t *testing.T) Client {
 	return c
 }
 
+func ptr[T any](t T) *T {
```

Comment:
Arg, the pipelines still run against some ancient versions that do not support generics, yet. I'll give these a bump in a separate PR.

Reply:
See https://github.com/mittwald/go-powerdns/pull/26